### ISSUE TITLE
How to feed CP(F1) population markers into TSPmap?

### ISSUE THREAD

**Issue by xiekunwhy** [mckaylab/TSPmap#6]
Hi,

Can you provide some examples of using tspmap to construct a genetic map of CP(F1) population?

Best wishes,
Kun

**Comment by greymonroe** [comment]
Hi Kun,
Have you calculated recombination frequencies between markers yet?
Thanks,
Grey

**Comment by xiekunwhy** [comment]
Dear Grey,

I have not calculate rf yet, I don't know how to feed markers into TSPmap.
Do you mean that I need to calculate rf first using some other tools like R/qtl?

Best wishes
Kun

**Comment by greymonroe** [comment]
Yes I would suggest calculating rf with R/qtl first then the TSPmap uses the rf for calculating marker order. If you need help loading into TSPmap let me know!

**Comment by xiekunwhy** [comment]
Dear Grey,

Thank you for your suggestion. But R/qtl is too slow and memory consuming, I have about ~40k bin(from GBS data, too many errors maybe), I think I need to try some other marker grouping tools.

Kun

**Comment by greymonroe** [comment]
I see, so you have already tried generating the rf and found it was too slow with R/QTL?

**Comment by xiekunwhy** [comment]
Yes, I wanted to use R/qtl to construct genetic map for my projects last few weeks, I have also tried onemap, it was slow and memory consuming too.
I just want too find a tool to construct ultra-dense(10k-500k) genetic maps, Lep-MAP3 is very fast, but just half of markers were grouped after grouping and single joining, so I think I need to use another tools to verified the results.

Kun

**Comment by LovellHAGSC** [comment]
I'm fairly certain that Lep-MAP3 and other programs like it need a reference genome, so that the dimensionality of the data can be cut down to local problems (e.g. within chromosomes). No matter the program, if you try to compute and store in memory a 40k -by- 40k marker rf matrix you'll need a bigger computer. 

The big question tho is, why do you need so many markers? If it is an experimental mapping population, I imagine the map will be saturated at a far lower number. Perhaps prune the dataset by LD first? Thats what I usually do. 
JL

**Comment by xiekunwhy** [comment]
@LovellHAGSC , 

Marker number never be saturating for contigs/scaffolds fragment anchoring, and for chimeric breaking.

Kun
